src/audio/mux/mux.c

```diff
@@ -648,12 +648,22 @@ static int mux_reset(struct comp_dev *dev)
 								source_list);
 			struct comp_buffer __sparse_cache *sink_c = buffer_acquire(sink);
 			int state = -1;
+			int direction = -1;
 
-			if (sink_c->sink)
+			if (sink_c->sink) {
 				state = sink_c->sink->state;
+				direction = sink_c->sink->direction;
+			}
```

Comment:
`mux_reset()` is currently also shared between mux and demux, whereas the echo reference functionality is only relevant for the demux, right? Would it maybe make sense to separate `mux_reset()` and `demux_reset()` and only modify the latter for this case? See https://github.com/thesofproject/sof/pull/6502/commits/02968c64c1e2dda1c369d844a118e0c81b8d9ccb for an example attempt to similarly untangle `mux_trigger()`

Reply:
done